src/NATS.Client.Core/Commands/CommandWriter.cs

```diff
@@ -500,10 +500,18 @@ internal async Task TestStallFlushAsync(TimeSpan timeSpan)
                             }
 
                             // don't mark the message as complete if we have more data to send
-                            if (totalSize + pending > sent)
+                            if (sendEx == null)
```

Comment:
I don't think this is the correct behavior.  `sent` is already set correctly if there was a `sendEx`; it acts as if all of the bytes were sent during an exception.

So the code that marks which commands were consumed afterwards is the same, whether there was a Send Exception or not.  Any command considered fully sent is consumed.

Lastly, the `sendEx` is thrown, which will cause this task to eventually get recreated, and any examined data will be available to be sent again